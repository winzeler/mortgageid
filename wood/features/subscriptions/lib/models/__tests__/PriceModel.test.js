const { PriceModel } = require('#features/subscriptions/lib/models/PriceModel');

describe('PriceModel', () => {
  describe('constructor()', () => {
    // ---------------------------------------------------------------------------------------------

    it('should properly display monthly prices', () => {
      const price = new PriceModel({
        unit_amount: 9999,
        currency: 'usd',
        interval: 'month',
        interval_count: 1,
      });

      expect(price.intervalPrice).toMatchSnapshot();
    });

    // ---------------------------------------------------------------------------------------------

    it('should properly display multi-interval prices', () => {
      const price = new PriceModel({
        unit_amount: 9999,
        currency: 'usd',
        interval: 'month',
        interval_count: 3,
      });

      expect(price.intervalPrice).toMatchSnapshot();
    });

    // ---------------------------------------------------------------------------------------------

    it('should properly display GBP', () => {
      const price = new PriceModel({
        unit_amount: 9999,
        currency: 'gbp',
        interval: 'month',
        interval_count: 1,
      });

      expect(price.intervalPrice).toMatchSnapshot();
    });

    // ---------------------------------------------------------------------------------------------

    it('should properly display euros', () => {
      const price = new PriceModel({
        unit_amount: 9999,
        currency: 'eur',
        interval: 'month',
        interval_count: 1,
      });

      expect(price.intervalPrice).toMatchSnapshot();
    });
  });
});
