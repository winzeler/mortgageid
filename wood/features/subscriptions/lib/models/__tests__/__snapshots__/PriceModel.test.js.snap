// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceModel constructor() should properly display GBP 1`] = `"£99.99 GBP / month"`;

exports[`PriceModel constructor() should properly display euros 1`] = `"€99.99 EUR / month"`;

exports[`PriceModel constructor() should properly display monthly prices 1`] = `"$99.99 USD / month"`;

exports[`PriceModel constructor() should properly display multi-interval prices 1`] = `"$99.99 USD every 3 months"`;
